src/components/templates/PosterPage/components/ShareModal/ShareModal.tsx

```diff
@@ -0,0 +1,101 @@
+import React, { useCallback, useMemo } from "react";
+import { Modal } from "react-bootstrap";
+import { useCopyToClipboard } from "react-use";
+import classNames from "classnames";
+import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
+import { faFacebook, faTwitter } from "@fortawesome/free-brands-svg-icons";
+
+import { DEFAULT_MAP_BACKGROUND } from "settings";
+
+import { PosterPageVenue } from "types/venues";
+
+import { WithId } from "utils/id";
+import { getFullVenueInsideUrl } from "utils/url";
+
+import { LinkButton } from "components/atoms/LinkButton";
+
+import {
+  getTitleTextForSharing,
+  getFacebookHref,
+  getTwitterHref,
+} from "./helpers";
+
+import "./ShareModal.scss";
+
+export interface ShareModalProps {
+  show: boolean;
+  onHide: () => void;
+  venue: WithId<PosterPageVenue>;
+}
+
+export const ShareModal: React.FC<ShareModalProps> = ({
+  show,
+  onHide,
+  venue,
+}) => {
+  const url = getFullVenueInsideUrl(venue.id);
+  const [{ value: isShowCopiedText }, copyToClipboard] = useCopyToClipboard();
```

Comment:
[NB] This reads better:

```suggestion
  const [{ value: hasCopiedText }, copyToClipboard] = useCopyToClipboard();
```

Reply:
https://github.com/sparkletown/sparkle/pull/1396/commits/2deaa63420083f3bd24823d446c4c10e3da4bc2a